services/space.controller.js

```diff
@@ -76,15 +82,18 @@ class SpaceServices extends eventEmiter{
         const {inDate, finDate, tag} = req.query
         const tags = tag ? tag.split(",") : null
         let response
+        const searchTerm = searchTermConstructor(req.query)
         try{
-            const foundResponse = await Space.find(searchTermConstructor(req.query))
+            const foundResponse = await Space.find({...searchTerm.searchTerm})
             .populate("dateReservedId", ["initialDate", "finalDate", "tenantId"])
             .populate("spaceTags", ["name","description"])
             .populate("faqs", ["question", "answer", "_id"])
             response = foundResponse
             if(inDate && finDate) response = filterSpaceByDate(foundResponse, inDate, finDate)
             if(tag) response = await filterSpaceByTag(response, tags)
-            res.status(200).json(response)
+            const maxPages = Math.ceil(response.length / searchTerm.pagination.limit)
+            const arr = response.slice(searchTerm.pagination.skip , searchTerm.pagination.skip + searchTerm.pagination.limit )
```

Comment:
Esta funcionalidad ya viene con mongoose https://medium.com/javascript-in-plain-english/simple-pagination-with-node-js-mongoose-and-express-4942af479ab2

Reply:
si la intentamos utilizar pero no nos funcionó porque nosotros luego de hacer el llamado a moongose hacemos otras validaciones que me modifican el número de elementos que se van a enviar, fue un error de diseño porque hacemos filtros basados en el populate en vez de trabajar directamente con los filtros de la base de datos. Pero llegados a este punto pensamos que esta sería una solución viable y nos funciona. Ahí nos quedó la lección para el futuro los diseños de nuestros modelos en mongoose cuando vayamos a hacer filtros.